lib/spack/spack/package.py

```diff
@@ -639,26 +639,53 @@ def do_fetch(self):
                     "Will not fetch %s." % self.spec.format('$_$@'), checksum_msg)
 
         self.stage.fetch()
+
+        ##########
+        # Fetch resources
+        resources = self._get_resources()
+        # FIXME : choose the unique name appropriately. Is there a function somewhere for the base name ?
+        pieces = [self.name, str(self.version), self.spec.dag_hash()]
+        for resource in resources:
+            resource_stage_folder = '-'.join(pieces + [resource.name])
```

Comment:
Here I construct a name for the staging folder of the resource with the pattern `<package-stage-folder-name>-<resource-name>`:
* would this be a good choice for the name?
* is the stage folder name for the main package stored somewhere (so to avoid manual construction)?